### ISSUE TITLE
parcel support 

### ISSUE THREAD

**Issue by ocavue** [ckeditor/ckeditor5#986]
This issue is a *feature request*.

It seems that ckeditor 5 only support webpack for now. Does ckeditor have any plan to support [parcel](https://parceljs.org/)?

BTW, if you are interesting to it, there is the error when I try to build ckeditor 5 framework with parcel:

```
$ ./node_modules/.bin/parcel index.html
Server running at http://localhost:1234
🚨  /my_project/node_modules/@ckeditor/ckeditor5-ui/src/componentfactory.js: /my_project/node_modules/@ckeditor/ckeditor5-ui/src/componentfactory.js: unknown Statement of type "ForOfStatement"
    at Emitter.Ep.explodeStatement (/my_project/node_modules/regenerator-transform/lib/emit.js:648:13)
    at /my_project/node_modules/regenerator-transform/lib/emit.js:344:12
    at Array.forEach (<anonymous>)
    at Emitter.Ep.explodeStatement (/my_project/node_modules/regenerator-transform/lib/emit.js:343:22)
    at Emitter.Ep.explode (/my_project/node_modules/regenerator-transform/lib/emit.js:298:40)
    at PluginPass.exit (/my_project/node_modules/regenerator-transform/lib/visit.js:122:15)
    at newFn (/my_project/node_modules/babel-traverse/lib/visitors.js:276:21)
    at NodePath._call (/my_project/node_modules/babel-traverse/lib/path/context.js:76:18)
    at NodePath.call (/my_project/node_modules/babel-traverse/lib/path/context.js:48:17)
    at NodePath.visit (/my_project/node_modules/babel-traverse/lib/path/context.js:117:8)
```

---

If you'd like to see this issue resolved please react with 👍 to this post.

**Comment by oleq** [comment]
> The CSS files are a bigger problem. We use PostCSS with a couple of standard plugins. However, we also needed to implement our own custom plugin for it. @oleq will tell you more what it does and whether it's possible to leave without it.

It's the [theme importer](https://github.com/ckeditor/ckeditor5-dev/blob/master/packages/ckeditor5-dev-utils/lib/styles/themeimporter.js#L44-L60) plugin we're talking about. 

Every single UI component of the editor comes with some CSS. But that's only the wireframe. To make it look like an actual button/dropdown/input, a theme is required. 

ATM we have only one theme https://github.com/ckeditor/ckeditor5-theme-lark. What the theme importer plugin does is glue the wireframe CSS files with the theme CSS files, that's all.

This importer plugin and other PostCSS essentials are wrapped in the [`getPostCssConfig` helper](https://github.com/ckeditor/ckeditor5-dev/blob/master/packages/ckeditor5-dev-utils/lib/styles/getpostcssconfig.js) which generates the PostCSS-readable configuration. 

At some point, you have to import `getPostCssConfig()` and pass its output to PostCSS in your bundler, e.g. 

```js
const { styles } = require( '@ckeditor/ckeditor5-dev-utils' );

...

somePostCssOptions: styles.getPostCssConfig( {
	themeImporter: {
		themePath: require.resolve( '@ckeditor/ckeditor5-theme-lark' )
	},
	minify: true
} )
```
though I'm not sure what does it look like in parcel.

**Comment by ocavue** [comment]
Thanks for your reply. Your explanations are really helpful.

**Comment by ThaddeusJiang** [comment]
@ocavue can you share your solution?

**Comment by ocavue** [comment]
@ThaddeusJiang I didn't find out a solution in the end so I switch to webpack. Sorry about that. 

I will keep this issue open in case someone figures it out in the future.

**Comment by ocavue** [comment]
Just did some work. I found out that parcel can't import `postcss-import`. Both `import PI from 'postcss-import'` and `const PI = 'postcss-import'` will cause `Uncaught TypeError: Cannot convert undefined or null to object` in chrome's console. 

I get this error because [`getPostCssConfig` use `postcss-import`](https://github.com/ckeditor/ckeditor5-dev/blob/3a28590f9c45303cedc3db53eeaa663f75267fa3/packages/ckeditor5-dev-utils/lib/styles/getpostcssconfig.js#L23). As what @oleq said, I have to deal with getPostCssConfig. 

Maybe https://github.com/parcel-bundler/parcel/issues/329 is related but I'm not sure.

**Comment by abhinavthapa1998** [comment]
Any updates regarding this? Trying to use CKEditor5 with parcel and it throws this error in the css files during the build. 
![image](https://user-images.githubusercontent.com/84438186/182340791-9d128180-d70b-43d3-acb2-519602751e4c.png)

**Comment by Reinmar** [comment]
All those issues will probably be resolved once #15502 hit npm (the project is in progress, all feedback greatly welcomed)
